src/sqlfluff/rules/L052.py

```diff
@@ -179,7 +179,7 @@ def _handle_semicolon_same_line(
             info.anchor_segment,
             info.whitespace_deletions,
             [
-                SymbolSegment(raw=";", type="symbol", name="semicolon"),
+                SymbolSegment(raw=";", type="statement_terminator"),
```

Comment:
Not in scope for this PR, but it seems like there should be a way to create this segment using the type alone (not requiring the hardcoded `";"`).

Reply:
yeah I agree - it's a bit messy. For now I think it's also out of scope but I'll keen an eye on it.